fiftyone/core/session.py

```diff
@@ -135,6 +155,20 @@ def __init__(self, dataset=None, view=None, port=5151, remote=False):
                 % (self.server_port, self.server_port, self.server_port)
             )
 
+    def __del__(self):
+        """Deletes the Session by removing it from the `_subscribed_sessions`
+        global and deleting (stopping) the associated
+        :class:`fiftyone.core.service.ServerService` if no other sessions are
+        subscribed.
+        """
+        global _subscribed_sessions  # pylint: disable=global-statement
+        _subscribed_sessions[self._port].discard(self)
+
+        if len(_subscribed_sessions[self._port]) == 0:
+            global _server_services  # pylint: disable=global-statement
+            if self._port in _server_services:
+                _server_services.pop(self._port)
```

Comment:
It would probably be a good idea to call `_server_services[self._port].stop()` first - `__del__` won't run right away, and won't run if something else has a reference to the service.